controllers/usersignup/usersignup_controller.go

```diff
@@ -756,6 +765,71 @@ func (r *Reconciler) ensureSpaceBinding(
 	return nil
 }
 
+// annotateCaptchaAssessment attempts to annotate the original assessment (if any) to provide feedback so that future scores can be improved
+// the attempt to annotate the previous assessment is a best-effort, we do not want to reconcile again if the request fails because captcha is a separate service
+// and should not have any impact to the UserSignup flows if there are issues with this request
+func (r *Reconciler) annotateCaptchaAssessment(ctx context.Context, userSignup *toolchainv1alpha1.UserSignup, newState string) {
+	logger := log.FromContext(ctx)
+	assessmentID, assessmentIDFound := userSignup.Annotations[toolchainv1alpha1.UserSignupCaptchaAssessmentIDAnnotationKey]
+	if !assessmentIDFound {
+		// there's no assessment ID to annotate
+		return
+	}
+
+	newAssessmentAnnotation := r.shouldAnnotateCaptchaAssessment(userSignup, newState)
+	if newAssessmentAnnotation == recaptchapb.AnnotateAssessmentRequest_ANNOTATION_UNSPECIFIED {
```

Comment:
Shouldn't we also check if the `newAssessmentAnnotation` is the same as the current one from the UserSignup resource? And if it's so then it was already submitted to recaptcha so we can/should ignore it now? Just to avoid duplicating the same assessment annotation for the same users multiple times.

Reply:
Yeah, good point. I think it technically shouldn't happen because it's called from the setStateLabel function which does the `if oldState == state ` check before continuing but it's a good idea to check the new assessment against the old one in case we end up moving the code around in the future.

Reply:
94e1342413e18a430f97c78dbf77314264d01ffe